File: app/src/main/java/com/sysu/edu/widget/DataStoreManager.java
package com.sysu.edu.widget;

import android.content.Context;

import androidx.datastore.preferences.core.Preferences;
import androidx.datastore.preferences.core.PreferencesKeys;
import androidx.datastore.preferences.rxjava3.RxPreferenceDataStoreBuilder;
import androidx.datastore.rxjava3.RxDataStore;


public class DataStoreManager {
    public static final Preferences.Key<String> TODAY_CLASS =
            PreferencesKeys.stringKey("today_class");
    private static RxDataStore<Preferences> dataStore = null;

    public static synchronized RxDataStore<Preferences> getInstance(Context context) {
        if (dataStore == null) dataStore = new RxPreferenceDataStoreBuilder(
                context.getApplicationContext(),
                "today_class"
        ).build();
        return dataStore;
    }
}

File: app/src/main/java/com/sysu/edu/widget/DataStoreManager.kt
package com.sysu.edu.widget

import android.content.Context
import androidx.datastore.preferences.core.Preferences
import androidx.datastore.preferences.core.stringPreferencesKey
import androidx.datastore.preferences.rxjava3.RxPreferenceDataStoreBuilder
import androidx.datastore.rxjava3.RxDataStore

object DataStoreManager {
	val TODAY_CLASS: Preferences.Key<String> = stringPreferencesKey("today_class")
	private var dataStore: RxDataStore<Preferences> ?= null
	@Synchronized fun getInstance(context: Context): RxDataStore<Preferences> {
		if (dataStore == null) dataStore = RxPreferenceDataStoreBuilder(context.applicationContext, "today_class").build()
		return dataStore!!
	}
}